salt/state.py

```diff
@@ -2507,22 +2508,16 @@ def _get_envs(self):
             envs.extend([x for x in list(self.opts['file_roots'])
                          if x not in envs])
         env_order = self.opts.get('env_order', [])
+        # Remove duplicates
+        env_order = list(collections.OrderedDict.fromkeys(env_order))
```

Comment:
list(set(env_order)) does not preserve the order. The used version is the shortest one for removing duplicates with preserving the order.